Cpufit/python/pycpufit/cpufit.py

```diff
@@ -0,0 +1,297 @@
+"""
+Program written from scratch following example of pyGPUfit by Jakub Dokulil (jakub.dokulil@imp.ac.at).
+
+For GPUfit see https://github.com/gpufit/Gpufit, http://gpufit.readthedocs.io/en/latest/bindings.html#python
+
+"""
+# %%
+
+import os
+import time
+from ctypes import cdll, POINTER, byref, c_int, c_float, c_char, c_char_p, c_size_t
+import numpy as np
+
+# %%
+package_dir = os.path.dirname(os.path.realpath(__file__))
+
+if os.name == 'nt':
+    lib_path = os.path.join(package_dir, 'cpufit.dll')  # library name on Windows
+elif os.name == 'posix':
+    lib_path = os.path.join(package_dir, 'libCpufit.so')  # library name on Unix
+else:
+    raise RuntimeError('OS {} not supported by pyGpufit.'.format(os.name))
+
+lib = cdll.LoadLibrary(lib_path)
+
+# %%
+cpufit_func = lib.cpufit

```

Comment:
May as well use cpufit_constrained. arguments (and argtypes) can then be copied from pygpufit.